changelog.d/11672.feature

```diff
@@ -0,0 +1,4 @@
+Whenever the `spamcheck` forbids a user to create a room, a `SynapseError` with `errcode` `M_UNKNOWN` is returned to the client. However, it would be more specific to return the `errcode` `M_UNKNOWN`.
+
+Therefore, this PR will raise `M_FORBIDDEN` instead of `M_UNKNOWN` whenever `synapse/events/spamcheck.py:user_may_create_room` returns `False`.
+Note: The `M_FORBIDDEN` code is set in the methods which currently call `user_may_create_room`, not in the method `user_may_create_room` itself.
```

Comment:
This changelog is quite long! I'm not sure if it's made clear or not, but it should really just be one bullet point (see the [changelog](https://github.com/matrix-org/synapse/blob/develop/CHANGES.md) for an idea).

I'm not the best writer in the world, but I would probably write it like this:
```suggestion
Return an `M_FORBIDDEN` error code instead of `M_UNKNOWN` when a spam checker module prevents a user from creating a room.
```